packages/url_launcher/url_launcher_android/test/url_launcher_android_test.dart

```diff
@@ -197,12 +198,198 @@ void main() {
+              webViewConfiguration:
+                  InAppWebViewConfiguration(enableJavaScript: false)));
+
+      expect(api.passedWebViewOptions?.enableJavaScript, false);
+    });
+
+    test('passes enableDomStorage to webview', () async {
+      final UrlLauncherAndroid launcher = UrlLauncherAndroid(api: api);
+      await launcher.launchUrl(
+          'http://example.com/',
+          const LaunchOptions(
+              mode: PreferredLaunchMode.inAppWebView,
+              webViewConfiguration:
+                  InAppWebViewConfiguration(enableDomStorage: false)));
+
+      expect(api.passedWebViewOptions?.enableDomStorage, false);
+    });
+
+    test('passes through no-activity exception', () async {
+      final UrlLauncherAndroid launcher = UrlLauncherAndroid(api: api);
+      await expectLater(
+          launcher.launchUrl('https://noactivity',
+              const LaunchOptions(mode: PreferredLaunchMode.inAppWebView)),
+          throwsA(isA<PlatformException>()));
+    });
+
+    test('throws if there is no handling activity', () async {
+      final UrlLauncherAndroid launcher = UrlLauncherAndroid(api: api);
+      await expectLater(
+          launcher.launchUrl('unknown://scheme',
+              const LaunchOptions(mode: PreferredLaunchMode.inAppWebView)),
+          throwsA(isA<PlatformException>().having(
+              (PlatformException e) => e.code, 'code', 'ACTIVITY_NOT_FOUND')));
+    });
+  });
+
+  group('launch with custom tab', () {
+    test('calls through', () async {
+      final UrlLauncherAndroid launcher = UrlLauncherAndroid(api: api);
+      final bool launched = await launcher.launchUrl('http://example.com/',
+          const LaunchOptions(mode: PreferredLaunchMode.inAppBrowserView));
+      expect(launched, true);
+      expect(api.usedWebView, true);
+      expect(api.allowedCustomTab, true);
+    });
+  });
+
+  group('launch with platform default', () {
+    test('uses custom tabs for http', () async {
+      final UrlLauncherAndroid launcher = UrlLauncherAndroid(api: api);
+      final bool launched = await launcher.launchUrl(
+          'http://example.com/', const LaunchOptions());
+      expect(launched, true);
+      expect(api.usedWebView, true);
+      expect(api.allowedCustomTab, true);
+    });
+
+    test('uses custom tabs for https', () async {
+      final UrlLauncherAndroid launcher = UrlLauncherAndroid(api: api);
+      final bool launched = await launcher.launchUrl(
+          'https://example.com/', const LaunchOptions());
+      expect(launched, true);
+      expect(api.usedWebView, true);
+      expect(api.allowedCustomTab, true);
+    });
+
+    test('uses external for other schemes', () async {
+      final UrlLauncherAndroid launcher = UrlLauncherAndroid(api: api);
+      final bool launched = await launcher.launchUrl(
+          'supportedcustomscheme://example.com/', const LaunchOptions());
+      expect(launched, true);
+      expect(api.usedWebView, false);
+    });
+  });
```

Comment:
Everything from here up is pretty much copies of the existing legacy (`launch`) tests, but using `launchUrl`.